tests/benchdnn/dnnl_common.cpp

```diff
@@ -1126,7 +1185,8 @@ static int check_total_size(
     }
 
     size_t total_size_cpu = check_mem_size_args.total_size_cpu;
-    if (is_cpu()) total_size_cpu += check_mem_size_args.total_size_device;
+    if (is_unified_test_memory())
+        total_size_cpu += check_mem_size_args.total_size_device;
```

Comment:
I would make it flipped, something like: `if(!is_discrete_gpu()) add-extra-size`
If some ID is missing/unknown - we better be on the safer side and take a pessimistic estimate. This would save time on debugging.
If the concern is that in this case we don't catch some missing device IDs - then I would incorporate a device IDs check similar to this one:
https://github.com/oneapi-src/oneDNN/blob/00ba39488366411d5bb9736aa576d34d3b9485f4/src/gpu/intel/jit/binary_format.cpp#L244

Reply:
I suggested to make a generic check for summed device + cpu memory against cpu memory regardless of the GPU type. This would allow to avoid much code with low risk of important cases being skipped.

Reply:
Agreed. I investigated the interface with differentiating between iGPU and dGPU as that is something we have encountered for GPU work in the past. Given the complexity required in solving this problem, I agree this simpler solution is better for now.